src/Renci.SshNet/Common/Pipe.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace Renci.SshNet.Common
+{
+    using System;
+    using System.IO;
+
+    /// <summary>
+    /// A generic pipe to pass through data.
+    /// </summary>
+    internal class Pipe : IDisposable
+    {
```

Comment:
As long as we don't leak this publicly, I don't mind. However, I still think the actual belongs in the PipeOutputStream, and this PipeOutputStream should be the only think that is passed to the PipeInputStream ctor.

If - big *if* - we do not want to dispose the Pipe by itself (meaning we may want to be able to dispose the writer without disposing the reader), then *Pipe* becomes an empty box.